### ISSUE TITLE
Allow generic attributes on HTML

### ISSUE THREAD

**Issue by bontaq** [purview-framework/purview#25]
Right now the attributes allowed on html are limited to:
```haskell
data Attributes action where
  OnClick :: ToJSON action => action -> Attributes action
  Style :: String -> Attributes action
```
In the spirit of more escape hatches, we should add something like
```haskell
data Attributes action where
  OnClick :: ToJSON action => action -> Attributes action
  Style :: String -> Attributes action
  Generic :: String -> String -> Attributes action
```

So that people can define their own.  The two important attributes I'm thinking of are `id` and `class`.  `class` would need some special handling but that could be another issue to open (joining them together with spaces).  For `id`, it could look something like this for testing:

```haskell
identifier string = Generic "id" string

component = identifier "hello" $ div [ text "it's a hello div" ]

main = run println component
```
and the html produced would look something like
```html
<div id="hello">it's a hello div</div>
```
This would unblock a bunch of stuff like setting attributes on inputs (`type="radio"`, `label="something"`, etc)

**Comment by bontaq** [comment]
Fixed by #29 , or at least fixed enough!
